### ISSUE TITLE
How to apply scaler from training set on test set

### ISSUE THREAD

**Issue by dorienh** [jdb78/pytorch-forecasting#1329]
I want to create training and test set. 

My training set is defined as below. I believe a scaler is applied by default. 

I though of creating a test_dataset using a similar command but with different cutoff. However, I would ideally want the the scaler learned on the training set to be applied for the test set. Otherwise there is a lookahead bias. Could this have anything to do with add_target_scales? I could not find much in doc or tutorial about this. Alternatively, I can scale the df before creating the dataset, but it would be nicer if there is a way... 

```
training_dataset = TimeSeriesDataSet(
    dataset[lambda x: x['idx'] <= training_cutoff],
    time_idx='idx', 
    target="y",
    group_ids=["filename"], # groups different time series
    min_encoder_length=1,  
    max_encoder_length=max_encoder_length,
    min_prediction_length=1, 
    max_prediction_length=max_prediction_length,
    static_categoricals=["filename"],
    static_reals=[],
    time_varying_known_categoricals=[],  
    time_varying_known_reals=['Open', 'High', 'Low', 'Close', 'Volume', 'ATR', 'CC', 'Close_pct' , 'Volume_pct'], ## add other variables later on
    time_varying_unknown_categoricals=['y'],
    time_varying_unknown_reals=['idx'], #what is unknown? remove idx?
    categorical_encoders={'filename': pytorch_forecasting.data.encoders.NaNLabelEncoder(add_nan=True)}, ## how are nans processed? there should be none. 
    # scalers= StandardScaler, Defaults to sklearn’s StandardScaler()
    add_relative_time_idx=True, 
    add_target_scales=True, ## what is this? 
    add_encoder_length=True,
    allow_missing_timesteps=False, # does not allow idx missing
    predict_mode = False
)
```

**Comment by dorienh** [comment]
It occurred to me I can use the below. Would this reuse the scaler fitted on the training set? 

 validation = training_dataset.from_dataset(training,
     data[lambda x: x.time_idx<=validation_cutoff], 
     min_prediction_idx=training_cutoff + 1)

 test = training_dataset.from_dataset(training, 
     data, 
     min_prediction_idx=validation_cutoff + 1)

**Comment by grosestq** [comment]
Hi,

Your solution is correct in my opinion.
All i know about "add_target_scales" is that it is used as static feature (you can see it when looking ah variable importance).

Best regards
